### PR TITLE
update `no-deprecated-destroyed-lifecycle` rule

### PR DESCRIPTION
**Pull request by Alanscut**
in Vue 3.x:
- `destroyed` lifecycle option has been renamed to `unmounted`
- `beforeDestroy` lifecycle option has been renamed to `beforeUnmount`
see in https://v3.vuejs.org/guide/migration/introduction.html#other-minor-changes.

so we should fix them instead of giving suggestion

### CODE REVIEW

File: tests/lib/rules/no-deprecated-destroyed-lifecycle.js [hunk 1]
```diff
@@ -71,8 +71,8 @@ ruleTester.run('no-deprecated-destroyed-lifecycle', rule, {
       code: `
       <script>
       export default {
-        beforeUnmount:beforeDestroy,
-        unmounted:destroyed,
+        beforeUnmount,
+        unmounted,
```
Comment (ota-meshi) [inline tests/lib/rules/no-deprecated-destroyed-lifecycle.js:75]: Why did you need to change this?

**Review by ota-meshi — APPROVED** [review]

**Review by ota-meshi — CHANGES_REQUESTED** [review]
I made a mistake.

File: tests/lib/rules/no-deprecated-destroyed-lifecycle.js (hunk 1 above)
Comment (Alanscut) [reply tests/lib/rules/no-deprecated-destroyed-lifecycle.js]: sorry, that's a misoperation, updated now

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!
